game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsFactoryTest.java

```diff
@@ -0,0 +1,131 @@
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsFactory.builder()
+                .battleActions(mock(BattleActions.class))
+                .returnFire(MustFightBattle.ReturnFire.ALL)
+                .diceRoller((arg1, arg2) -> new DiceRoll())
+                .firingGroupSplitter(
+                    (arg1) ->
+                        FiringGroup.groupBySuicideOnHit(
+                            "army", List.of(givenAnyUnit()), List.of(mock(Unit.class))))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(names, is(generalFightStepStrings(attacker, defender, "army")));
+  }
+
+  private List<String> getStepNames(final List<BattleStep> steps) {
+    return steps.stream().flatMap(step -> step.getNames().stream()).collect(Collectors.toList());
+  }
+
+  @Test
+  @DisplayName("With two groups, there should be two sets of fire round strings")
+  void getNamesWithTwoFiringGroups() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsFactory.builder()
+                .battleActions(mock(BattleActions.class))
+                .returnFire(MustFightBattle.ReturnFire.ALL)
+                .diceRoller((arg1, arg2) -> new DiceRoll())
+                .firingGroupSplitter(
+                    (arg1) ->
+                        List.of(
+                            FiringGroup.groupBySuicideOnHit(
+                                    "army", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0),
+                            FiringGroup.groupBySuicideOnHit(
+                                    "spies", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                .get(0)))
+                .casualtySelector((arg1, arg2) -> new CasualtyDetails())
+                .side(OFFENSE)
+                .battleState(
+                    FakeBattleState.givenBattleStateBuilder()
+                        .attacker(attacker)
+                        .defender(defender)
+                        .build())
+                .build()
+                .createSteps());
+
+    assertThat(
+        names,
+        is(
+            mergeSteps(
+                generalFightStepStrings(attacker, defender, "army"),
+                generalFightStepStrings(attacker, defender, "spies"))));
+  }
+
+  @Test
+  @DisplayName("With the default name UNITS, the step names should be readable")
```

Comment:
What is meant by 'readable'?

Reply:
The "step names" is shown to the user in the Battle UI.  So "readable" would be understandable to the user.  When I first looked at the resulting strings in the UI, it sounded odd to me so I worked it a little to make it more "readable".  But it is very subjective.

Reply:
Ah, qualifying 'readable' as 'human readable' should make that clear. 'readable' can be understood in different ways.